File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        List<Character> lst = new ArrayList<>();
        int ans = 0;
        for(int i = 0 ; i < s.length(); i++){
            if(lst.contains(s.charAt(i))){
                int idx = lst.indexOf(s.charAt(i));
                lst.remove(idx);
                if(idx>0)
                    lst.subList(0,idx).clear();
            }
            lst.add(s.charAt(i));
            ans = Math.max(ans, lst.size());
        }
        return ans;
    }
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.php
class Solution {

    /**
     * @param String $s
     * @return Integer
     */
    function lengthOfLongestSubstring($s) {
        $n = strlen($s);
        $map = array();
        $ans = $i = 0;
        for ($j = 0; $j < $n; $j++) {
            if (array_key_exists($s[$j],$map)) {
                $i = max($map[$s[$j]], $i);
            }
            $ans = max($ans, $j - $i + 1);
            $map[$s[$j]] = $j + 1;
        }
        return $ans;
    }

}